catalog/to-consul/syncer.go

```diff
@@ -169,31 +194,31 @@ func (s *ConsulSyncer) watchReapableServices(ctx context.Context) {
 			return
 		}
 
-		// Update our blocking index
-		opts.WaitIndex = meta.LastIndex
```

Comment:
So I *think* understand what's happening here. Previously, this function (after the first run through) would block when it called `s.Client.Catalog().Services(&opts)` and wait for updates due to this WaitIndex here. Now that you've broken up that call into one per namespace, we can't do the blocking because it's different per namespace.

Is there not a way to make the catalog services call for all namespaces? Then we could keep the previous blocking behaviour.

If not, can you explain why this change in behaviour is okay.